Yoast/wordpress-seo
# P1 341 abstract dismissable alert

## Context


* Recently a generic way to dismiss Alert's was added. This PR adds a Container component for the existing Alert component that makes it easy to add a permanent dismissable Alert to our projects.

## Summary


This PR can be summarized in the following changelog entry:

* Adds the possibility to insert permanent dismissable Alerts.

## Relevant technical choices:

* For the sake of simplicity dismissal of Alerts within add-ons will be stored to the Yoast SEO store as well.
* The prop isn't added to the store for Taxonomies, nor is the wpseoApi / yoast-seo-api enqueued there since it is very unlikely that we will need this Alert over there. Can still be added later on when it is needed.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Test this branch in conjunction with the P1-341 branch of the MonoRepo
* Before testing make sure that the alertKey with which you are testing is added to the allowed keys. This can be achieved the easiest by activating the latest Video SEO RC as well and using that alertKey for testing
* Make sure that the Alertkey you're using isn't dismissed yet, the easiest way is to delete the `_yoast_alerts_dismissed` row form your `wp_usermeta` row in your database
* Add a `permanentDismissableAlert` in two places, for instance the `MetaboxFill` and `SidebarFill`
```
import PersistentDismissableAlert from"../../containers/PersistentDismissableAlert";
<PersistentDismissableAlert alertKey="video-editor-reactification-alert" type="info">Testalert</PersistentDismissableAlert>
```
* Make sure the permanent dismissable Alert looks like a normal dismissable Alert
* Hit the Dismiss button (cross) and make sure the Alert disappears
* * Make sure it has disappeared in the other instance as well
* Refresh your browser and make sure the Alert is still away

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* There are no integrations of this Container yet, so for now please make sure that existing Alerts still function like before, for instance: the upsell Alert within the Facebook preview on Free and the Alert on Settings -> Search appearance -> General that's shown when no Company name and logo are shown.

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes P1-341
